.github/workflows/pull_request.yml

```diff
@@ -88,8 +113,27 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
     - uses: actions/checkout@v4
+      # only run if code changes
+      if: env.run_tests == 'true'
```

Comment:
You can't test the variable before you created it !

You have to put this stop under the one where you compute run_tests (that's why your tests are currently failing)